### PR TITLE
Add PlayerTakeLecternBookEvent

### PR DESCRIPTION
**Pull request by TheLimeGlass**
### Description
Add Spigot's PlayerTakeLecternBookEvent

---
**Target Minecraft Versions:** 1.14+
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 1]
```diff
@@ -731,6 +732,16 @@ public class SimpleEvents {
 				)
 				.since("2.7");
 
+		if (Skript.isRunningMinecraft(1, 14))
+			Skript.registerEvent("Player Take Lectern", SimpleEvent.class, PlayerTakeLecternBookEvent.class, "[player] tak(e|ing [a]) (lectern book|book from [a] lectern)")
```
Comment (Moderocky) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java:736]: I'm a bit concerned about the name being "take lectern" when it's actually taking a book, not taking a whole lectern.

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 2]
```diff
@@ -731,6 +732,16 @@ public class SimpleEvents {
...
+					.description("Called when a player clicks the button to take a book of a Lectern. If this event is cancelled the book remains on the lectern.")
```
Comment (Moderocky) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java:737]:
```suggestion
					.description("Called when a player clicks the button to take a book from a lectern. If this event is cancelled the book remains on the lectern.")
```

**Review by Moderocky — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/events/SimpleEvents.java (hunk 1 above)
Comment (TheLimeGlass) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:736]: It's taken from the name of the event, to keep familiarity to the event.

File: src/main/java/ch/njol/skript/events/SimpleEvents.java (hunk 2 above)
Comment (TheLimeGlass) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:737]: Description copied from Spigot JavaDoc.

File: src/main/java/ch/njol/skript/events/SimpleEvents.java (hunk 1 above)
Comment (sovdeeth) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:736]: > It's taken from the name of the event, to keep familiarity to the event.

It needs to include the `book` part too. The event is Taking a Lectern Book, not Taking a Lectern.

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 3]
```diff
@@ -1753,6 +1755,21 @@ public ItemStack get(PlayerStonecutterRecipeSelectEvent event) {
...
+			EventValues.registerEventValue(PlayerTakeLecternBookEvent.class, ItemStack.class, new Getter<ItemStack, PlayerTakeLecternBookEvent>() {
+				@Override
+				public ItemStack get(PlayerTakeLecternBookEvent event) {
+					return event.getBook();
+				}
+			}, EventValues.TIME_NOW);
+			EventValues.registerEventValue(PlayerTakeLecternBookEvent.class, Lectern.class, new Getter<Lectern, PlayerTakeLecternBookEvent>() {
+				@Override
+				public Lectern get(PlayerTakeLecternBookEvent event) {
+					return event.getLectern();
```
Comment (ShaneBeee) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:1768]: Correct me if I'm wrong, but is Lectern.class not a blockstate?
Skript doesn't use BlockStates.
Shouldn't this return the block instead?
```suggestion
			EventValues.registerEventValue(PlayerTakeLecternBookEvent.class, Block.class, new Getter<Block, PlayerTakeLecternBookEvent>() {
				@Override
				public Block get(PlayerTakeLecternBookEvent event) {
					return event.getLectern().getBlock();
```

**Review by ShaneBeee — CHANGES_REQUESTED** [review]
small thing I noticed

**Comment by TheLimeGlass** [comment]
Not working on this anymore

**Pull request closed without merging**